x/adminmodule/keeper/proposal_legacy.go

```diff
@@ -29,16 +28,36 @@ func (k Keeper) SubmitProposalLegacy(ctx sdk.Context, content govv1beta1types.Co
 	}
 
 	headerTime := ctx.BlockHeader().Time
-
 	// submitTime and depositEndTime would not be used
 	proposal, err := govv1beta1types.NewProposal(content, proposalID, headerTime, headerTime)
 	if err != nil {
-		return govv1beta1types.Proposal{}, err
+		return govv1beta1types.Proposal{}, sdkerrors.Wrap(err, "failed to create proposal struct")
 	}
 
+	handler := k.RouterLegacy().GetRoute(proposal.ProposalRoute())
+	cacheCtx, writeCache := ctx.CacheContext()
+
+	// The proposal handler may execute state mutating logic depending
+	// on the proposal content. If the handler fails, no state mutation
+	// is written and the error message is returned.
+	err = handler(cacheCtx, content)
+	if err == nil {
+		proposal.Status = govv1beta1types.StatusPassed
+
+		// The cached context is created with a new EventManager. However, since
+		// the proposal handler execution was successful, we want to track/keep
+		// any events emitted, so we re-emit to "merge" the events into the
+		// original Context's EventManager.
+		ctx.EventManager().EmitEvents(cacheCtx.EventManager().Events())
+
+		// write state to the underlying multi-store
+		writeCache()
+	} else {
+		return govv1beta1types.Proposal{}, err
```

Comment:
so did I get it right that we won't have any proposals in the adminmodule state but the `Passed` ones? the failed proposals won't be recorded because such transactions won't succeed and no state mutations will happen at the adminmodule side, all of them will only be visible from the Neutron DAO's side?